rfcs/react-components/non-core-components.md

```diff
@@ -0,0 +1,388 @@
+- Composed of multiple primitive components
+- Specializations of primitive components
+
+We need to formalise how we will deliver these components not only as first party packages but also second party
+packages with our partners who might need more situational components powered by V9. This RFC refers to these kinds
+of components as `non-core`.
+
+## Problem statement
+
+We create components that will not be shipped as a part of our 'core' components under the `@fluentui/react-components`
+library but other packages. We will have to finalize the following requirements:
+
+- Should non-core components be able to deviate from V9 core concepts?
+- Should non-core components have a faster breaking change cadence?
+- Should non-core components live in a separate repo?
+- Should non-core components still be under the `@fluentui` scope?
+- How should users consume non-core packages?
+- How should non-core components be documented?
+- How should code for partner owned components be hosted?
+
+The Proposal treats each of the above questions separately and provides pros/cons for different approaches we could take.
+Keep in mind that some decisions will influence other decisions.
+
+None of the approaches are particularly complex to understand, but the resulting work could be complex. It is important
+that we figure out what answers for these questions so that we can start to move forward
+with technical implementation by following concrete requirements.
+
+## Detailed Design or Proposal
+
+### Deviating from v9 core concepts
+
+It's difficult to formalize what 'deviating' means here, and any attempt to do so will probably not be
+very future proof as the library continues to evolve. The main objective of this section is to decide whether or not
+we should allow deviations from v9 core concepts (whatever they are or might be in the future). Here are some
+examples of current deviations that could happen based on the current situation:
+
+- Allowing shorthand collections
+- 'Fatter' JSX elements that have more functionality
+- Less customizable styles
+- More opinionated layout
+
+The short list above indicates some possibilities to have components that are less flexible in cases where
+components are more specialized versions of core primitives that we don't want to maintain a large customization
+feature set for, or product/scenario specific components where we want more alignment across Microsoft.
+
+The extent of 'deviating' will have to depend on the requirements of each component, but we should be ready to accept
+that we might end up delivering a set of components that don't follow the principles that drive the core library
+(@fluentui/react-components).
+
+In cases where components need to deviate, we should notify the team through specs and architecture meetings
+(i.e. tech sync), what deviations are done and the context behind them. All non-core components should still
+follow the V9 spec driven approach.
+
+#### Pros
```

Comment:
i'm on team 'earned differences', where components should follow v9 approach (slots, hooks, child elements over render arrays) as much as possible. When users hit friction and want to deviate, we should discuss that friction and determine if that scenario is a valid earned difference, or maybe we just need better guidance, or a refined way to accomplish a goal.

example, user REALLY wants render arrays because that is what their customers are used to. Maybe we propose a way to create an exported function or hook that provides the same API while still leveraging the opt in/pay for play nature of v9.

Reply:
That's a really good point, I've incorporated this to the RFC.